multiaddr_test.go

```diff
@@ -166,6 +166,7 @@ func TestBytesToString(t *testing.T) {
 	testString("/ip4/127.0.0.1/udp/1234", "047f0000011104d2")
 	testString("/ip4/127.0.0.1/tcp/4321", "047f0000010610e1")
 	testString("/ip4/127.0.0.1/udp/1234/ip4/127.0.0.1/tcp/4321", "047f0000011104d2047f0000010610e1")
+	testString("/onion/aaimaq4ygg2iegci:80", "bc030010c0439831b48218480050")
```

Comment:
what is the port? is it always strictly tcp? or is it an onion port? (meaning the transport is abstracted out) 
If the client has to know the protocol, then I would imagine this being better: `/onion/<hash>/tcp/80`